Description: Handle test_pad_constant failures on multiple archs
Bug-Origin: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=983758
Author: Alastair McKinstry <mckinstry@debian.org>
Last-Updated: 2021-09-01
Forwarded: no

--- a/xarray/tests/test_dataarray.py
+++ b/xarray/tests/test_dataarray.py
@@ -20,6 +20,9 @@
     broadcast,
     set_options,
 )
+from . import (
+    arm_xfail,
+)
 from xarray.coding.times import CFDatetimeCoder
 from xarray.convert import from_cdms2
 from xarray.core import dtypes
@@ -4518,6 +4521,7 @@
             out = da.polyfit("x", 8, full=True)
             np.testing.assert_array_equal(out.polyfit_residuals.isnull(), [True, False])
 
+    @pytest.mark.xfail
     def test_pad_constant(self):
         ar = DataArray(np.arange(3 * 4 * 5).reshape(3, 4, 5))
         actual = ar.pad(dim_0=(1, 3))
@@ -4546,6 +4550,7 @@
             with pytest.raises(ValueError, match="cannot convert float NaN to integer"):
                 ar.pad(x=1, constant_values=np.NaN)
         else:
+            # XFAIL on arm64,i386 as the constants don't overflow to NaN
             actual = ar.pad(x=1, constant_values=np.NaN)
             expected = xr.DataArray(
                 [-9223372036854775808, 9, -9223372036854775808], dims="x"
